### DISCUSSION TITLE
Updated docs to show checking for errors

### DISCUSSION THREAD

**Discussion by richardltc** [pocketbase/pocketbase#196] [Q&A] [answered]
Hi,

As a SvelteKit and JS newbie, I'd really appreciate it if the docs (Creating a user) could be update to show the actual code for checking for error/the result reported back by PocketBase?

Thank you

Richard

**Comment by ganigeorgiev** [comment, marked as answer]
The SDK is framework agnostic so it doesn't have anything specific to SvelteKit.
All JS sdk services (with an exception to `Realtime`) returns a standard JavaScript [Promise object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise). So in general there are 2 ways to check for errors:

```js
client.Records.getOne("demo", "RECORD_ID")
  .then((record) => {
      // success...
      console.log(record);
  })
  .catch((err) => {
      console.log(err);
  });
```

Or if you are using the `async/await` syntax:

```js
try {
    const record = await client.Records.getOne("demo", "RECORD_ID");
} catch (err) {
    console.log(err);
}
```

As mentioned in the [Caveats section in the JS-SDK README](https://github.com/pocketbase/js-sdk#caveats), all returned errors are normalized as `ClientResponseError` with the following properties and methods:
```js
ClientResponseError {
    url:           string,     // requested url
    status:        number,     // response status code
    data:          { ... },    // the api json error response (this is probably what you are looking for!)
    isAbort:       boolean,    // is abort/cancellation error
    originalError: Error|null, // the original non-normalized error
}
```

**Reply by richardltc** [reply]
Sorry for the newbie questions. I'm getting: `ClientResponseError 400: Failed to create user` with the following code:

`   import PocketBase from "pocketbase";

    const client = new PocketBase("http://localhost:8090");

    let loading = false;
    let email = "";
    let password = "";

    const createUser = async () => {
        try {
            const user = await client.Users.create({
                email: email,
                password: password,
                passwordConfirm: password,
            });
        } catch (err) {
            console.log(err);
        }
`
I guess I'm still doing something wrong?

Richard

**Reply by ganigeorgiev** [reply]
The error is probably because probably some of the validations for email or password doesn't pass.
To inspect the error and show for example a message to the user, you can try to access the `err.data` object - it contains the error api json response. eg.
```js
console.log(err.data);

// where err.data usually have the following format:
// see the "Response 400" tab in https://pocketbase.io/docs/api-users/ > "Create user section"
{
  "code": 400,
  "message": "An error occurred while submitting the form.",
  "data": {
    "email": {
      "code": "validation_required",
      "message": "Missing required value."
    }
  }
}
```

**Reply by richardltc** [reply]
Perfect! Tank you :)

**Reply by agingaRichard** [reply]
Having the same problem here. How did you fix it?

**Comment by jimafisk** [comment]
@ganigeorgiev apologies if this is something you're still working on, but I'm testing out [v0.8.0-rc1](https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0-rc1) and I'm having trouble getting the specific error with `console.log(err.data);` inside a `try`/`catch` block. 

I want to provide validation, for example if someone tries to submit a blank form. This was working previously with a `$client.Users.create()`, but now if I try to do a  `$client.collection('members').create()` I get a generic `message` of `Failed to create record` but `data` appears to be blank. The logs in the PB Admin appear to provide more context though:

```
{
  "errorDetails": "DrySubmit error: email: cannot be blank; password: cannot be blank; passwordConfirm: cannot be blank.",
  "errorMessage": "Failed to create record."
}
```

I appreciate any guidance, thank you and congrats on the `v0.8.0-rc1`launch it looks awesome!

**Reply by ganigeorgiev** [reply]
This is a bug/side-effect of the errors wrapping in the new release - https://github.com/pocketbase/pocketbase/blob/rc/apis/record_crud.go#L237

I'll fix it either as part of the next rc2 or the final v0.8.0 release (I'll have to also update the tests to check the returned error fields).

**Reply by jimafisk** [reply]
This is working again in [v0.8.0-rc2](https://github.com/pocketbase/pocketbase/releases/tag/v0.8.0-rc2), thank you @ganigeorgiev!
